### PR TITLE
Convert AnimationUtils functions to extension functions.

### PR DESCRIPTION
**Pull request by Isira-Seneviratne**
#### What is it?
- [ ] Bugfix (user facing)
- [ ] Feature (user facing)
- [x] Codebase improvement (dev facing)
- [ ] Meta improvement to the project (dev facing)

#### Description of the changes in your PR

- Convert the static methods defined in `AnimationUtils` to Kotlin extension functions.

#### APK testing 

[debug.zip](https://github.com/TeamNewPipe/NewPipe/files/5812225/debug.zip)

#### Due diligence
- [x] I read the [contribution guidelines](https://github.com/TeamNewPipe/NewPipe/blob/HEAD/.github/CONTRIBUTING.md).

### CODE REVIEW

File: app/src/main/java/org/schabi/newpipe/fragments/list/comments/CommentsFragment.java [hunk 1]
```diff
@@ -84,16 +84,14 @@ public void showLoading() {
...
 
         if (!result.getErrors().isEmpty()) {
             showSnackBarError(result.getErrors(), UserAction.REQUESTED_COMMENTS,
                     NewPipe.getNameOfService(result.getServiceId()), result.getUrl(), 0);
         }
 
-        if (disposables != null) {
-            disposables.clear();
-        }
+        disposables.clear();
```
Comment (XiangRongLin) [inline app/src/main/java/org/schabi/newpipe/fragments/list/comments/CommentsFragment.java:94]: removing the null check won't cause problems?

**Review by XiangRongLin — COMMENTED** [review]
Did you test everything that you changed? Since there are only very little unit tests, refactoring like this requires extensive manual testing.

File: app/src/main/java/org/schabi/newpipe/fragments/list/comments/CommentsFragment.java (hunk 1 above)
Comment (Isira-Seneviratne) [reply app/src/main/java/org/schabi/newpipe/fragments/list/comments/CommentsFragment.java:94]: The variable is final, so there won't be any problems.

**Comment by Isira-Seneviratne** [comment]
> Did you test everything that you changed? Since there are only very little unit tests, refactoring like this requires extensive manual testing.

I made sure to preserve the existing behavior.

Just to be sure, I tested it out and didn't find any issues.

**Review by Stypox — DISMISSED** [review]
Looks good to me, thank you :-)

File: app/src/main/java/org/schabi/newpipe/ktx/TextView.kt [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+import android.animation.ArgbEvaluator
+import android.animation.ValueAnimator
+import android.util.Log
+import android.widget.TextView
+import androidx.annotation.ColorInt
+import androidx.interpolator.view.animation.FastOutSlowInInterpolator
+import org.schabi.newpipe.BuildConfig
+
+private const val TAG = "TextViewUtils"
+private const val DEBUG = BuildConfig.BUILD_TYPE != "release"
```
Comment (TobiGr) [inline app/src/main/java/org/schabi/newpipe/ktx/TextView.kt:16]: Please import the DEBUG const from MainActivity, so we only need to change one var when there is the need to change the debug logic.

File: app/src/main/java/org/schabi/newpipe/ktx/View.kt [hunk 3]
```diff
@@ -0,0 +1,325 @@
...
+import androidx.annotation.FloatRange
+import androidx.core.view.ViewCompat
+import androidx.core.view.isGone
+import androidx.core.view.isInvisible
+import androidx.core.view.isVisible
+import androidx.interpolator.view.animation.FastOutSlowInInterpolator
+import org.schabi.newpipe.BuildConfig
+
+private const val TAG = "ViewUtils"
+private const val DEBUG = BuildConfig.BUILD_TYPE != "release"
```
Comment (TobiGr) [inline app/src/main/java/org/schabi/newpipe/ktx/View.kt:22]: same here

**Review by TobiGr — DISMISSED** [review]
Small thing

**Comment by Stypox** [comment]
Oh, then also `TAG` should be changed to `= CurrentJavaClass.class.getSimpleName()`

File: app/src/main/java/org/schabi/newpipe/ktx/TextView.kt (hunk 2 above)
Comment (Isira-Seneviratne) [reply app/src/main/java/org/schabi/newpipe/ktx/TextView.kt]: > Oh, then also `TAG` should be changed to `= CurrentJavaClass.class.getSimpleName()`

There doesn't seem to be a way to get a reference to the generated class for top-level declarations.

**Comment by Isira-Seneviratne** [comment]
> Oh, then also `TAG` should be changed to `= CurrentJavaClass.class.getSimpleName()`

There doesn't seem to be a way to get a reference to the generated class for top-level declarations.

**Review by Stypox — APPROVED** [review]
Thank you. @TobiGr can this be merged?

**Comment by Stypox** [comment]
I tested many places and everything seems to work well

**Pull request merged**